### PR TITLE
Add "mentorias" command

### LINKED ISSUE #12: Add ability to list mentors and their details
## Context

As we discussed, the first feature we want to add on Bino is to make information regarding mentors available.

We want users to be able to list all mentors and see their availabity schedule and focused areas.

Our initial proposal for the commands is:

```
> mentorias

- Lucia
- Marcus
- Julio
- Camila
```

And to get details about a specific mentor:

```
> mentoria Lucia

Lucia

Posso te ajudar com desenolvimento pessoal, saúde e bem-estar e prática do inglês.

Horários:
  - Seg: 09:00-10:00, 13:00-13:45
  - Ter: 14:30-15:00
  (...)
```

## Out of scope:

- We'll start with static data, no need to update list of mentors and/or their data
- Collecting any information regarding the requester

## Open questions:

- How we do automated testing?
- Do we expect to use bino on Caravana's discord service for testing?

### PR DESCRIPTION
**Pull request by fernando-alves**
I've started to put together some structure to add `mentoria` as command. I wanted to gather feedback as soon as possible so I decided to open this PR.

I'll keep the notes/description updated as it evolves.

Related to #12

### CODE REVIEW

**Comment by fernando-alves** [comment]
I think I'm done with this for now. I've added the command and a proposal to structure it. Mentor only needed name, so I kept it simple.

So far commands need both:
- `CanHandle`: Given the message, the command decides if the command should answer to it or not
- `Process`: Given the message, the command returns a string representing what should be returned to the user

I don't have much experience with go, so please share any feedback if stuff is not written "the go way".

Here's some other points:

- Updated `README.md` to include command to run tests
- Removed all example code from `main.go` and left only the command handler
- Docker setup is still needs some work. The app fails to start cause the database server, even though has started, is still not accepting connections. I think we might need some additional check along with `depends_on`.
- As far as I understood the event we are using, `discordgo.MessageCreate`, is used for both DMs and messages in channels where the bot is included. We might need to adjust it so only DMs and direct mentions like `@bino mentorias` are processed.

And some questions:

- [ ] I thought the compiler would enforce comment for public members of the packages, is it not the case anymore?
- [ ] Does it make sense to have the commands and struct that implement and interface? Again, I'm not familiar with go.
- [ ] I image the safety check on `Process` would probably be used by all commands, is there a better way to share this behaviour?
```{golang}
func (m Mentorship) Process(message string) (string, error) {
	if !m.CanHandle(message) {
		return "", fmt.Errorf("Can't process message '%s'", message)
	}

	(...)
}
```

**Comment by fernando-alves** [comment]
After some feedback I've made a few changes:

- Commands are executed from `CommandRunner` this way we can enforce the `CanRun` check out-of-the-box
- Renamed interfaces so they follow convention and end with `-er` 
- Introduced a `bot` entity so responsibilities of identifying, running and handling errors from commands are separated from handling sessions and messages from discord, done in `CommandMessageHandler`. Also helps with testability.

File: commands/mentorship.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+var Mentorship = CommandRunner{
+	Command: mentorshipCommand,
+}
+
+func canRun(message string) bool {
+	return message == "mentorias"
+}
+
+func run(message string) (string, error) {
+	var stringBuffer bytes.Buffer
```
Comment (marcuxyz) [inline commands/mentorship.go]: here, you can use this [named return values](https://tour.golang.org/basics/7)

Comment (marcuxyz) [reply commands/mentorship.go]: you can use `fmt.Sprintf()` to interpolation variables inside string and return string.

File: discord/bot.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	return bot{
+		mentorship: mentorship,
+	}
+}
+
+func (b bot) Process(message string) string {
+	if strings.EqualFold(message, "mentorias") {
+		response, err := b.mentorship.Run(message)
+		if err != nil {
+			fmt.Printf("Error while responding to membership message: %e", err)
```
Comment (marcuxyz) [inline discord/bot.go]: you use the `log package`  only for logging message in **standard output**, instead of `fmt.Printf`

**Review by marcuxyz — COMMENTED** [review]
please reconsider the comments

File: discord/bot.go (hunk 2 above)
Comment (fernando-alves) [reply discord/bot.go]: Good point, fixed!

File: commands/mentorship.go (hunk 1 above)
Comment (fernando-alves) [reply commands/mentorship.go]: Fixed!

This is interesting, when do you think we should be using named returns? Honestly I did not even remember it was a thing :P

**Pull request closed without merging**